sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -218,7 +247,14 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       bucketSpec = getBucketSpec,
       options = extraOptions.toMap)
 
-    dataSource.write(mode, df)
+    val destination = source match {
+      case "jdbc" => extraOptions.get(JDBCOptions.JDBC_TABLE_NAME)
+      case _ => extraOptions.get("path")
```

Comment:
> `outputParams` The output parameters in case the method is invoked as a result of a write operation.

It sounds like `OutputParams` is designed for the write path. It is being used for description? Could we make it more general? For example, using a Map[String, String] like data structure? In the future, we can just add the tag we need?